### PR TITLE
add error handling for when no arguments passed for options at the end of command

### PR DESCRIPTION
**Pull request by jacobmealey**
master currently segfaults if you run `urler --url`, this is a quick fix so let me know if there should be more careful considerations taken. Note that this is different from #16 not sure what is different from testing but mine still seg faults if no URL passed.

### CODE REVIEW

File: urler.c [hunk 1]
```diff
@@ -279,7 +279,8 @@ int main(int argc, const char **argv)
     else if(argv[0][0] == '-' && argv[0][1] == '-') {
       /* dash-dash prefixed */
       int usedarg = 0;
-      if(getlongarg(&o, argv[0], argv[1], &usedarg))
+      
+      if(argc < 2 || getlongarg(&o, argv[0], argv[1], &usedarg))
```
Comment (bagder) [inline urler.c:283]: Doesn't this prevent the options that don't take arguments from working if specified last?

Comment (jacobmealey) [reply urler.c]: Oh woops! my bad. I just pushed another revision that should work better. I have it checking if the argument is null in `getlongarg()` It looks like this is something that should be checked for all arguments that take options. for example `urler --url github.com --append-path`

**Comment by jacobmealey** [comment]
Okay, I added checks for when the user runs commands that end with `--append-path` `--append-url` and `--url` but don't provide text. 

ps sorry for the many commits and if this PR is noisy

**Comment by bagder** [comment]
This still leaves out several other `---set-*` options from the check. I propose we maybe do it this way instead: #21

**Comment by bagder** [comment]
Lots of other internal changes but this was fixed in f64e450612740a1

**Pull request closed without merging**